src/compiler/nvcc.rs

```diff
@@ -743,19 +865,38 @@ where
             ),
             // cicc and ptxas are cacheable
             Some("cicc") => {
-                match compilation_flag.to_str() {
-                    // Fix for CTK < 12.8:
-                    // If `nvcc` is invoked with `-c` (or any of its variants), remove the
-                    // `--gen_module_id_file` flag. In this mode, we instruct `cudafe++`
-                    // to generate this file, so cicc shouldn't generate it again.
-                    Some("-c") | Some("--compile") | Some("-dc") | Some("--device-c")
-                    | Some("-dw") | Some("--device-w") => {
-                        if let Some(idx) = args.iter().position(|x| x == &gen_module_id_file_flag) {
-                            args.splice(idx..(idx + 1), []);
-                        }
+                // Fix for CTK < 12.0:
+                // Remove `--gen_module_id_file` if cudafe++ already does it
+                if cudafe_has_gen_module_id_file_flag {
+                    if let Some(idx) = args.iter().position(|x| x == &gen_module_id_file_flag) {
+                        args.splice(idx..idx + 1, []);
                     }
-                    _ => {}
                 }
+                // Add these flags if they're missing:
+                // * `--gen_c_file_name test_a.compute_XX.cudafe1.c`
+                // * `--stub_file_name test_a.compute_XX.cudafe1.stub.c`
+                // * `--gen_device_file_name test_a.compute_XX.cudafe1.gpu`
+                //
+                // This ensures the same `cicc` command is generated regardless
+                // of whether the compilation flag is `-c`, `-ptx`, or `-cubin`
+
+                // e.g. test_a.compute_XX.cpp1.ii
+                let mut nidx = args.len() - 3;
+                let name = args[nidx].clone();
+                // test_a.compute_XX.cpp1.ii -> test_a.compute_XX
+                let name = name.split(".cpp1.ii").next().unwrap();
+
+                for (flag, name) in [
+                    (&gen_c_file_name_flag, format!("{name}.cudafe1.c")),
```

Comment:
Could you elaborate the signficance of the `cudafe1.*` file extension? This smells like very application specifc